src/main/java/seedu/address/ui/Calender.java

```diff
@@ -0,0 +1,189 @@
+package seedu.address.ui;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.Year;
+import java.time.YearMonth;
+
+import javafx.fxml.FXML;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.Border;
+import javafx.scene.layout.BorderStroke;
+import javafx.scene.layout.BorderStrokeStyle;
+import javafx.scene.layout.BorderWidths;
+import javafx.scene.layout.CornerRadii;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+
+public class Calender extends UiPart<Region> {
+    private static final String FILEPATH = "Calender.fxml";
+    private LocalDate now;
+    private int year;
+    private Month month;
+    private Month headerMonth;
+    private int headerYear;
+    private boolean isLeapYear;
+
+    @FXML
+    private GridPane calenderGrid;
+    @FXML
+    private Label monthYearInput;
+
+    /**
+     * Represents the calender that is going to be displayed.
+     */
+    public Calender() {
+        super(FILEPATH);
+        this.now = LocalDate.now();
+        this.year = now.getYear();
+        this.month = now.getMonth();
+        this.headerMonth = now.getMonth();
+        this.headerYear = this.year;
+        this.isLeapYear = now.isLeapYear();
+        loadNow();
+    }
+
+    private int numOfDays() {
```

Comment:
Maybe we can use verb for this method name e.g. `getNumOfDays()`? (just to maintain consistency)